src/contracts/GPv2Settlement.sol

```diff
@@ -137,6 +137,119 @@ contract GPv2Settlement is GPv2Signing, ReentrancyGuard, StorageAccessible {
         emit Settlement(msg.sender);
     }
 
+    /// @dev Settle a single order directly with on-chain liquidity. This
+    /// function is provided as a "fast-path" for settlements without any
+    /// coincidence of wants.
+    ///
+    /// This type of settlement always assumes the order will be filled in full,
+    /// and while it does accept partially fillable orders, they must have no
+    /// prior filled amount, and will be executed in full.
+    ///
+    /// @param order The single order to settle.
+    /// @param signingScheme The signing scheme used for the order.
+    /// @param signature The signature of the order used to recover the owner
+    /// address.
+    /// @param transfers Direct transfers of user funds to execute to interact
+    /// with on-chain liquidity.
+    /// @param interactions Smart contract interaction to perform with executed
+    /// sell amount of the order.
+    function settleOrder(
+        GPv2Order.Data memory order,
+        Scheme signingScheme,
+        bytes calldata signature,
+        GPv2AllowanceManager.Transfer[] calldata transfers,
+        GPv2Interaction.Data[] calldata interactions
+    ) external nonReentrant onlySolver {
+        GPv2TradeExecution.Data memory executedTrade;
+        bytes memory orderUid;
+        {
+            (bytes32 orderDigest, address owner) =
+                recoverOrderSigner(order, signingScheme, signature);
```

Comment:
Can you check that this computation does not break the unicity of the orderDigest (therefore of the uid)? Since we use assembly to compute the hash, it might be that the extra bytes in the unpacked calldata could change the hash without changing the component of the orders, like in the test in `describe("uid uniqueness"...`.

Reply:
Great point!